projects/demo/src/modules/components/tag/examples/4/index.ts

```diff
@@ -0,0 +1,26 @@
+import {Component} from '@angular/core';
+import {changeDetection} from '@demo/emulate/change-detection';
+import {encapsulation} from '@demo/emulate/encapsulation';
+
+@Component({
+    selector: 'tui-tag-example-4',
+    templateUrl: './index.html',
+    styleUrls: ['./index.less'],
+    changeDetection,
+    encapsulation,
+})
+export class TuiTagExample4 {
+    tags: readonly string[] = [
+        'Taiga UI',
+        'is an open-source library',
+        'for awesome people',
+    ];
+
+    handleTagEdited(newTagCaption: string, tagIndex: number) {
+        if (newTagCaption) {
```

Comment:
```ts
handleTagEdited(newTagCaption: string, tagIndex: number) {
    if (newTagCaption) {
        this.tags = [...this.tags].splice(tagIndex, 1, newTagCaption);
    } else {
        this.tags = [...this.tags].splice(tagIndex, 1);
    }
}
```

we can try a little less, but not the point, we do not consider from the point of view of optimization

```ts
handleTagEdited(newCaption: string, currentIndex: number) {
   this.tags = this.tags.map((caption, index) => index === currentIndex ? newCaption : caption);
}
```

Reply:
🤔 my solution seems to be more clear, I think maybe wait for someone to also discuss this with us?

Reply:
anyway, it's just my recommendation, don't worry about that)

Reply:
But I think there is a bug in your code

![image](https://user-images.githubusercontent.com/12021443/150943441-109522de-ceee-4e8e-bfce-22ba57d7e25f.png)

![image](https://user-images.githubusercontent.com/12021443/150943517-f9792727-d2ad-42b5-ac3e-879c4806072e.png)

because we need the new array to consist of the same elements, except for the replacement

Reply:
Ooops yeah, I have a bug here.

Reply:
I would write `map(...).filter(Boolean)`. It's not optimal, but very concise.

Reply:
yea, I prefer variant from @waterplea